tests/includes/t51-test-case.php

```diff
@@ -0,0 +1,31 @@
+<?php
+declare( strict_types=1 );
+
+namespace Team51\Test;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Includes utility functions useful for testing Team51 CLI code.
+ */
+
+abstract class T51TestCase extends TestCase {
+
+	/**
+	 * Invokes a private or protected method on an object.
+	 *
+	 * @param   object  $object       The object on which to invoke the method.
+	 * @param   string  $method_name  The name of the method to invoke.
+	 * @param   array   $parameters   The parameters to pass to the method.
+	 *
+	 * @return  mixed   The return value of the method.
+	 */
+
+	public function invoke_method( &$object, $method_name, array $parameters = array() ) {
```

Comment:
How would this work if we wanted to test a static method on a class that wasn't instantiated?  Would there be a recommended other method to use, or a way to override this?

Reply:
I'm still on that learning curve, and so far I haven't found a way since we need to pass an object to instantiate the ReflectionClass. Those static methods would ideally be public, at least if we want to test them.
This brings up a greater philosophical question - ie should we even be testing private or protected methods? From what I understand, the answer is generally no. If that's the case then we're going to have a difficult time unit testing the cli tool, and maybe should look into command level testing, though I feel that's not helpful for finding issues in the individual methods within a class/command.
Does this mean we should/could be using public methods instead? (Rhetorical, kind of, opinions welcomed).

Reply:
Honestly, if it's something we'd like tested, I'm 100% in favor of just making it public.